flocker/provision/_gce.py

```diff
@@ -431,44 +435,31 @@ class GCEInstanceBuilder(PClass):
     project = field(type=unicode)
     compute = field()
 
-    def create_instance(self, instance_name, image=None, **kwargs):
+    def _execute_create_instance(self, instance_name, image=None, **kwargs):
         """
-        Create a GCE instance.
+        Executes a create instance request on GCE and returns the operation.
 
-        :param unicode instance: The name of the instance to create.
-        :param unicode image: The link to the GCE image to use for the base
-            disk. Defaults to the latest CentOS 7 image if unspecified.
-        :param **kwargs: Other keyword arguments for the creation of the
-            instance. See documentation of ``_create_gce_instance_config`` for
-            keyword argument values.
-        :returns GCEInstance: The instance that was started.
+        :returns dict: A GCE operation resource object, suitable for use with
+            ``wait_for_operation``.
         """
         if image is None:
             image = get_latest_gce_image_for_distribution('centos-7',
                                                           self.compute)
-        config = _create_gce_instance_config(
+        body = _create_gce_instance_config(
             instance_name=instance_name,
             project=self.project,
             zone=self.zone,
             image=image,
             **kwargs
         )
-        operation = self.compute.instances().insert(
+
+        return self.compute.instances().insert(
             project=self.project,
             zone=self.zone,
-            body=config
+            body=body
         ).execute()
 
-        operation_result = wait_for_operation(
-            self.compute, operation, timeout_steps=[5]*60)
-
-        if not operation_result:
-            raise ValueError(
-                "Timed out waiting for creation of VM: {}".format(
-                    instance_name
-                )
-            )
-
+    def _make_gce_instance(self, instance_name):
```

Comment:
:question: Maybe add a docstring here explaining that this function generates a flocker.provisioning shaped record of the GCEInstance from resource queries to the GCE API.